omnigibson/object_states/temperature.py

```diff
@@ -38,8 +38,12 @@ def update_temperature_from_heatsource_or_sink(cls, objs, temperature, rate):
             rate (float): Heating rate of the source / sink
         """
         # Get idxs for objs
-        idxs = [cls.OBJ_IDXS[obj] for obj in objs]
-        cls.VALUES[idxs] += (temperature - cls.VALUES[idxs]) * rate * og.sim.get_rendering_dt()
+        idxs = np.zeros(len(objs), dtype=int)
+        for i, obj in enumerate(objs):
+            # No need to explicitly add objects to updated objects set here, since @self.global_update() automatically
+            # handles this which occurs after all heatsources are updated
+            idxs[i] = cls.OBJ_IDXS[obj]
```

Comment:
are these not equivalent to the original code?

Reply:
+1